test/remote/gateways/remote_cyber_source_rest_test.rb

```diff
@@ -70,6 +70,88 @@ def test_successful_purchase
     assert_nil response.params['_links']['capture']
   end
 
+  def test_successful_refund
+    purchase = @gateway.purchase(@amount, @visa_card, @options)
+    response = @gateway.refund(@amount, purchase.authorization, @options)
+
+    assert_success response
+    assert response.test?
+    assert_equal 'PENDING', response.message
+    assert response.params['id'].present?
+    assert response.params['_links']['void'].present?
+  end
+
+  def test_failure_refund
+    purchase = @gateway.purchase(@amount, @card_without_funds, @options)
+    response = @gateway.refund(@amount, purchase.authorization, @options)
+
+    assert_failure response
+    assert response.test?
+    assert_match %r{Declined - One or more fields in the request contains invalid data}, response.params['message']
+    assert_equal 'INVALID_DATA', response.params['reason']
+  end
+
+  def test_successful_less_than_the_payment_amount_refund
+    purchase = @gateway.purchase(@amount, @visa_card, @options)
+    response = @gateway.refund(@amount / 2, purchase.authorization, @options)
+
+    assert_success response
+    assert response.test?
+    assert_equal 'PENDING', response.message
+    assert response.params['id'].present?
+    assert response.params['_links']['void'].present?
+  end
+
+  def test_successful_more_than_the_payment_amount_refund
```

Comment:
Shouldn't we expect this to fail?

Reply:
I guess if their sandbox simulates the delay in confirming, we wouldn't, and just get a pending result.

Reply:
At this moment I have no way to prove that a refund of greater value than the purchase is complete (even if it should fail), I get to the pending response therefore I will remove this test